### PR TITLE
Cp0 config

### PR DESCRIPTION
**Pull request by Bozydarek**
This branch adds function which prints the configuration registers values, together with their description. (Issue #10)

In the case of bit value - 0 or 1 means respectively false (disabled, not present, etc) or true (enabled, present, etc).

In other cases, I put a description of the meaning of the respective values ( in square brackets ).

### CODE REVIEW

**Comment by Bozydarek** [comment]
Now I'm stuck because:
* cache size is 0 and i don't know that is my fault or is it true ?
* I can't find in documentation a clear answer to the question about min and max size pages
* I don't even know what else is interesting in these configure registers

Moreover I think that I should add 1 to mips_cpuinfo.ic_nways and mips_cpuinfo.dc_nways

File: bitmap.h [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #define BIT_CLR(n, b) ((n) &= ~(1 << (b)) )
 #define BIT_INV(n, b) ((n) ^=  (1 << (b)) )
 #define BIT_GET(n, b) (((n) & (1<<(b))) != 0)
+#define BITS_GET(n, l, b) ( ((n) & (( (1<<(l)) - 1) << (b) )) >> b)
```
Comment (cahirwpz) [inline bitmap.h]: Please reimplement it as a [assembly inline](https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html) using `EXT` instruction. Implement `BITS_PUT` with `INS` instruction as well.

Comment (cahirwpz) [inline bitmap.h]: Oops... actually those are defined in `mips-mti-elf/include/mips/mips32.h` from toolchain directory, please use them.

File: main.c [hunk 2]
```diff
@@ -58,7 +73,66 @@ void udelay (unsigned usec)
...
     }
 }
-
+/* 
+ * Read configuration registers value, its interpretation 
+ * and save it into the struct mips_cpuinfo
+ */
+void read_config()
+{
+    int config = mfc0(16, 0);
```
Comment (cahirwpz) [inline main.c]: Please use symbolic names for CP0 registers. Look at `mips-mti-elf/include/mips/m32c0.h` in the toolchain directory.

**Comment by cahirwpz** [comment]
Please `git rebase` your branch on top of current master.

File: main.c [hunk 3]
```diff
@@ -70,6 +86,75 @@ void mdelay (unsigned msec)
     while(final > clock_get_ms());
 }
 
+/* 
+ * Read configuration registers value, its interpretation 
+ * and save it into the struct mips_cpuinfo
+ */
+void read_config()
+{
+    int config = mfc0(16, 0); // CP0 Register 16, Select 0
```
Comment (cahirwpz) [inline main.c]: `mips/m32c0.h` contains following definition:

```
#define mips32_getconfig0() mips32_get_c0(C0_CONFIG0)
```

**Pull request merged**